test/test_sparse.py

```diff
@@ -2177,6 +2177,20 @@ def test_factory_empty_indices(self):
         expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+    def test_sparse_abs(self):
+        device = 'cuda' if self.is_cuda else 'cpu'
+        i = torch.tensor([[0, 1, 1],
+                          [2, 0, 2],
+                          [2, 0, 2]], dtype=torch.long, device=device)
+
+        v = torch.tensor([3, -44, -5], dtype=torch.float, device=device)
+
+        s = torch.sparse_coo_tensor(i, v, torch.Size([2, 4, 4]), device=device)
+
+        dense_tensor = s.to_dense()
+        self.assertEqual(s.abs().to_dense(), dense_tensor.abs())
+        self.assertEqual(s.abs().device, dense_tensor.abs().device)
+
```

Comment:
See above comments for additional test coverage required. While we'll take a PR for abs that manually implements this behavior, future unary ops should be implemented generically. Ideally we would create a "sparse_op_db", analogous to "op_db", that can be used to enumerate the sparse ops. If you're interested in helping with this, @hlahkar, please let me know.

Reply:
I can help in the development of the Sparse Test Framework as per your suggestion; please let me know how we can cooperate on the same

Reply:
Thank you! Sorry for the delayed response. I responded on a different issue with a suggestion.